client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx

```diff
@@ -0,0 +1,360 @@
+		const { domain, selectedSite, translate } = this.props;
+
+		if ( domain.expired || domain.isSubdomain || ! domain.isEligibleForInboundTransfer ) {
+			return null;
+		}
+
+		const path = domainTransferIn( selectedSite.slug, domain.name, true );
+
+		return (
+			<VerticalNavItem path={ path }>
+				{ translate( 'Transfer Domain to WordPress.com' ) }
+			</VerticalNavItem>
+		);
+	}
+
+	resolveStatus() {
+		const { domain, translate, moment } = this.props;
+		const { expiry } = domain;
+
+		if ( isExpiringSoon( domain, 30 ) ) {
+			const expiresMessage = translate( 'Expires in %(days)s', {
+				args: { days: moment.utc( expiry ).fromNow( true ) },
+			} );
+
+			if ( isExpiringSoon( domain, 5 ) ) {
+				return {
+					statusText: expiresMessage,
+					statusClass: 'status-error',
+					icon: 'info',
+				};
+			}
+
+			return {
+				statusText: expiresMessage,
+				statusClass: 'status-warning',
+				icon: 'info',
+			};
+		}
+
+		if ( ! domain.pointsToWpcom ) {
+			return {
+				statusText: translate( 'Action required' ),
+				statusClass: 'status-error',
+				icon: 'info',
+			};
+		}
+
+		return {
+			statusText: translate( 'Active' ),
+			statusClass: 'status-success',
+			icon: 'check_circle',
+		};
+	}
+
+	renderExpiringSoon() {
+		const { domain, translate, moment } = this.props;
+		const { expiry } = domain;
+
+		if ( ! isExpiringSoon( domain, 30 ) ) {
+			return null;
+		}
+
+		if ( domain.bundledPlanSubscriptionId ) {
+			return (
+				<div>
+					<p>
+						{ translate(
+							'Your domain mapping will expire with your plan in {{strong}}%(days)s{{/strong}}. Please renew your plan before it expires or it will stop working.',
+							{
+								components: {
+									strong: <strong />,
+								},
+								args: {
+									days: moment.utc( expiry ).fromNow( true ),
+								},
+							}
+						) }
+					</p>
+					<RenewButton
+						primary={ true }
+						selectedSite={ this.props.selectedSite }
+						subscriptionId={ parseInt( domain.bundledPlanSubscriptionId, 10 ) }
+						customLabel={ translate( 'Renew your plan ' ) }
+						tracksProps={ { source: 'mapped-domain-status', mapping_status: 'expiring-soon-plan' } }
+					/>
+				</div>
+			);
+		}
+		return (
+			<div>
+				<p>
+					{ translate(
+						'Your domain mapping will expire in {{strong}}%(days)s{{/strong}}. Please renew it before it expires or it will stop working.',
+						{
+							components: {
+								strong: <strong />,
+							},
+							args: {
+								days: moment.utc( expiry ).fromNow( true ),
+							},
+						}
+					) }
+				</p>
+				<RenewButton
+					primary={ true }
+					selectedSite={ this.props.selectedSite }
+					subscriptionId={ parseInt( domain.subscriptionId, 10 ) }
+					tracksProps={ { source: 'mapped-domain-status', mapping_status: 'expiring-soon' } }
+				/>
+			</div>
+		);
+	}
+
+	renderSettingUpNameservers() {
+		const { domain, selectedSite, translate } = this.props;
+		if (
+			selectedSite.jetpack ||
```

Comment:
Let's maybe use the selectors to figure this out. Like:
https://github.com/Automattic/wp-calypso/blob/2de5e432fc90d3793d8bac4f8aa3e7a7dfb5432c/client/state/selectors/is-site-automated-transfer.js#L13

Otherwise I'm afraid at some point someone will add some more logic to that selector and we won't be in sync with it. I'm sure there's a similar one for Jetpack.